src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -504,7 +504,7 @@ public override async Task<long> RunCompactionAsync(long target)
                     this.GetElapsedCompactionMilliseconds());
 
                 var tokenSource = new CancellationTokenSource();
-                var timeoutTask = Task.Delay(TimeSpan.FromMinutes(10), tokenSource.Token); //.ContinueWith(_ => tokenSource.Dispose());
+                var timeoutTask = Task.Delay(TimeSpan.FromMinutes(10), tokenSource.Token).ContinueWith(_ => tokenSource.Dispose());
```

Comment:
I don't think this is the right way to dispose the timer task.

The problem is that you want to dispose it after the timeout completes AND after line 526. Otherwise line 526 may throw an exception. 

That is why I suggested to put the ContinueWith on line 529.